waiter/integration/waiter/autoscaling_test.clj

```diff
@@ -241,18 +242,16 @@
           _ (log/info "making canary request...")
           {:keys [cookies service-id]} (request-fn)
           get-target-instances
-          (fn []
-            (let [instances
-                  (loop [routers (routers waiter-url)
```

Comment:
This was just a stylistic change I made while I was editing the test anyway. `loop` feels very imperative, making the code feel less functional.